### PR TITLE
[mono][jit] Patch jump trampolines on amd64/arm64 after a method has …

### PR DESCRIPTION
**Pull request by vargaz**
…been compiled.

Part of the fix for https://github.com/dotnet/runtime/issues/96710.

### CODE REVIEW

**Review by lambdageek — APPROVED** [review]

**Comment by lambdageek** [comment]
@vargaz should we have some kind of flag not to do this if we're running on executable only pages? (I'm not sure if any of our platforms do that for JITed pages - maybe Android?)

**Comment by bholmes** [comment]
I think this also improves the icall scenario  that is listed in https://github.com/dotnet/runtime/issues/96710.

File: src/mono/mono/mini/tramp-amd64.c [hunk 1]
```diff
@@ -596,7 +628,11 @@ mono_arch_create_specific_trampoline (gpointer arg1, MonoTrampolineType tramp_ty
 
 	code = buf = (guint8 *)mono_mem_manager_code_reserve_align (mem_manager, size, 1);
 
-	if (((gint64)tramp - (gint64)code) >> 31 != 0 && ((gint64)tramp - (gint64)code) >> 31 != -1) {
+	if (is_jump_tramp) {
+		size += 64;
+		/* Align the trampoline so the first instruction doesn't cross a cache line */
+		code = buf = (guint8 *)mono_mem_manager_code_reserve_align (mem_manager, size, 16);
+	} if (((gint64)tramp - (gint64)code) >> 31 != 0 && ((gint64)tramp - (gint64)code) >> 31 != -1) {
```
Comment (bholmes) [inline src/mono/mono/mini/tramp-amd64.c:635]: are you missing an newline or an else?  any reason this is on the same line?

**Pull request merged**